05/app.js

```diff
@@ -14,44 +14,39 @@ function Student(name, surname) {
     }
     
-    //Metoda przyjmuje jeden parametr subject. Uzyskuje dostep do tablicy ocen dla danego przedmiotu z obiektu 'grades' przez 'this.grades[subject].
-    //Sprawdza czy tablica ocen dla przedmiotu istnieje.
-    //Jesli istnieja oceny dla danego przedmiotu, metoda oblicza ich srednia arytmetyczna przez funkcje reduce.
-    //Wynik jest zaokraglony do dwoch miejsc po przecinku, korzystajac z metody 'toFixed(2).
-    Student.prototype.calculateSubjectAverage = function (subject) {
-        const subjectGrades = this.grades[subject];
-        if (!subjectGrades || subjectGrades.length === 0) {
-            return 0;
-        }
-        const getAverageGrade = subjectGrades.reduce((a, b) => a + b, 0) / subjectGrades.length;
-        return getAverageGrade.toFixed(2);
-    }
+    /*
+        Jesli podam parametr 'subject' funkcja obliczy srednia ocen tylko dla konkretnego przedmiotu.
+        Jesli nie podam parametru 'subject' funkcja obliczy srednia ze wszystkich ocen studenta.
 
-    //Metoda przyjmuje jeden parametr i sprawdza czy istnieja jakiekolwiek przedmioty z ocenami w obiekcie grades.
-    //Uzyskuje liste nazw przedmiotow jako tablice kluczy obiektu 'grades' za pomoca 'Object.keys(this.grades).
-    //Jesli nie ma zadnych przedmiotow metoda zwraca 0.00 jako wynik.
-    //Jesli istnieja oceny metoda inicjuje zmienne totalSum oraz totalGrades.
-    //Metoda iteruje przez liste przedmiotow za pomcoa forEach i sprawdza czy istnieja oceny dla kazdego z przedmiotow.
-    //Jesli tak iteruje przez oceny w danym przedmiocie za pomoca kolejnego forEach i sumuje je w zmiennej totalSum oraz zwieksza 'totalGrades'.
-    //Po zakonczeniu metoda sprawdza czy istnieja jakiekolwiek oceny. Jesli nie ma zadnych zwraca '0.00'.
-    //Wynik ogolnej sredniej oceny jest zaokraglany do dwoch miejsc po przecinku dzieki 'toFixed(2).
-    Student.prototype.calculateOverallAverage = function () {
+        
+    */ 
+    Student.prototype.getAverageGrade = function (subject) {
         const subjects = Object.keys(this.grades);
         if (subjects.length === 0) {
             return '0.00';
         }
         let totalSum = 0;
         let totalGrades = 0;
     
-        subjects.forEach((subject) => {
+        if (subject) {
             const subjectGrades = this.grades[subject];
             if (subjectGrades && subjectGrades.length > 0) {
-                subjectGrades.forEach((grade) => {
+                subjectGrades.forEach(grade => {
                     totalSum += grade;
                     totalGrades++;
                 });
             }
-        });
+        } else {
+            subjects.forEach((subject) => {
+                const subjectGrades = this.grades[subject];
+                if (subjectGrades && subjectGrades.length > 0) {
+                    subjectGrades.forEach((grade) => {
+                        totalSum += grade;
+                        totalGrades++;
+                    });
```

Comment:
Tutaj zwróciłbym uwagę, że ten kod powtarza się z kodem 34-26 - już tego bym nie zmieniał, ale docelowo lepiej zamienić powielony kod na funkcje i uruchamiać ją w obu miejscach. Aby to zrobić wygodnie pewnie lepie byłoby zapisywać wszystkie oceny w tablicy i dopiero obliczać średnią. U Ciebie jest od razu suma oraz ilość elementów.